File: controller/controller.js
import exportToExcel from "../utils/exporter.js"
import Searcher from "../model/searcher.js"

export default class Controller {
    constructor (window) {
        this.searcher = new Searcher(this)
        this.mainWindow = window
    }
    
    async search(dados) {
        try {
            const resultBusca = await this.searcher.search(dados)
            return {'resultado': resultBusca, 'contagem': this.searcher.contagem}
        }
        catch(error) {
            console.log(error)
            return {'resultado': false, 'contagem': 0}
        }
    }

    export(path) {
        try {
            exportToExcel(path, this.searcher)
            
            return true
        }
        catch (error) {
            console.log(error)
            return false
        }
    }

    atualizar_progresso(atual, final) {
        this.mainWindow.webContents.send('atualizar-progresso', [atual, final])
    }
}

File: controller/controller.py
from model import searcher

class Controller:

    def __init__(self, searcher):
        self.searcher_controll = searcher
    
    def search(self, dados) -> object:
        try:
            self.searcher = searcher.Searcher(dados=dados, controller=self)
            busca = self.searcher.search()
            return {'resultado': busca, 'contagem': self.searcher.contagem}
        except AttributeError as error:
            print(error)
            return {'resultado': False, 'contagem': 0}

    def export(self, path) -> bool:
        try:
            self.searcher.export_to_excel(path)
            return True
        except:
            return False

    def atualizar_progresso(self, atual, final):
        self.searcher_controll.atualizar_progress(atual=atual, final=final)